### PR TITLE
pkh-flow fixes

### PR DESCRIPTION
**Pull request by rbrtrfl**
in the scope of commit

fix(pkh-flow): fix navigation, guards and doneFunctions around staatliche-leistungen

it is required to change the name of the select variable from `staatlicheLeistungenPKH` to `staatlicheLeistungen` in `/prozesskostenhilfe/formular/finanzielle-angaben/einkuenfte/staatliche-leistungen` in Strapi

### CODE REVIEW

File: app/flows/prozesskostenhilfeFormular/finanzielleAngaben/doneFunctions.ts [hunk 1]
```diff
@@ -66,23 +63,33 @@ export const wertsachenDone: ProzesskostenhilfeFinanzielleAngabenGuard = ({
...
+          einkuenfteDone({ context }) &&
+          partnerDone({ context }) &&
+          eigentumDone({ context }) &&
+          kinderDone({ context }) &&
+          eigentumZusammenfassungDone({ context }) &&
+          andereUnterhaltszahlungenDone({ context }) &&
+          ausgabenDone({ context }) &&
+          ausgabenZusammenfassungDone({ context })
+        );
+    }
```
Comment (Spencer6497) [inline app/flows/prozesskostenhilfeFormular/finanzielleAngaben/doneFunctions.ts:85]: nit: i think a ternary would be more appropriate here, given that there are only two cases, z.B.
```
return context.staatlicheLeistungen === 'asylbewerberleistungen' || context.staatlicheLeistungen === 'grundsicherung' ? true : <doneFunctions>
```

File: app/flows/prozesskostenhilfeFormular/finanzielleAngaben/doneFunctions.ts [hunk 2]
```diff
@@ -14,9 +14,8 @@ type ProzesskostenhilfeFinanzielleAngabenGuard =
 export const partnerDone: ProzesskostenhilfeFinanzielleAngabenGuard = ({
   context,
 }) =>
-  hasGrundsicherungOrAsylbewerberleistungen({
-    context,
-  }) ||
+  (context.staatlicheLeistungen != undefined &&
+    hasStaatlicheLeistungen({ context })) ||
   ["no", "widowed"].includes(context.partnerschaft ?? "") ||
```
Comment (Spencer6497) [inline app/flows/prozesskostenhilfeFormular/finanzielleAngaben/doneFunctions.ts:19]: `hasStaatlicheLeistungen` tests for _any_ staatlicheLeistungen, but I think the only case in which we skip over Partner is Grundsicherung or Asylbewerberleistungen

File: app/flows/prozesskostenhilfeFormular/finanzielleAngaben/doneFunctions.ts [hunk 3]
```diff
@@ -26,17 +25,15 @@ export const partnerDone: ProzesskostenhilfeFinanzielleAngabenGuard = ({
 export const kinderDone: ProzesskostenhilfeFinanzielleAngabenGuard = ({
   context,
 }) =>
-  hasGrundsicherungOrAsylbewerberleistungen({
-    context,
-  }) ||
+  (context.staatlicheLeistungen != undefined &&
+    hasStaatlicheLeistungen({ context })) ||
```
Comment (Spencer6497) [inline app/flows/prozesskostenhilfeFormular/finanzielleAngaben/doneFunctions.ts:29]: see above

File: app/flows/prozesskostenhilfeFormular/finanzielleAngaben/doneFunctions.ts [hunk 4]
```diff
@@ -26,17 +25,15 @@ export const partnerDone: ProzesskostenhilfeFinanzielleAngabenGuard = ({
...
   context.hasKinder == "no" ||
   arrayIsNonEmpty(context.kinder);
 
 export const andereUnterhaltszahlungenDone: ProzesskostenhilfeFinanzielleAngabenGuard =
   ({ context }) =>
-    hasGrundsicherungOrAsylbewerberleistungen({
-      context,
-    }) ||
+    (context.staatlicheLeistungen != undefined &&
+      hasStaatlicheLeistungen({ context })) ||
```
Comment (Spencer6497) [inline app/flows/prozesskostenhilfeFormular/finanzielleAngaben/doneFunctions.ts:36]: see above

File: app/flows/prozesskostenhilfeFormular/finanzielleAngaben/einkuenfte/doneFunctions.ts [hunk 5]
```diff
@@ -110,7 +113,8 @@ export const furtherIncomeDone: ProzesskostenhilfeFinanzielleAngabenEinkuenfteGu
 
 export const einkuenfteDone: ProzesskostenhilfeFinanzielleAngabenEinkuenfteGuard =
   ({ context }) =>
-    hasGrundsicherungOrAsylbewerberleistungen({ context }) ||
+    (context.staatlicheLeistungen != undefined &&
+      hasStaatlicheLeistungen({ context })) ||
```
Comment (Spencer6497) [inline app/flows/prozesskostenhilfeFormular/finanzielleAngaben/einkuenfte/doneFunctions.ts:117]: see other comments but I'm not sure we should be skipping all einkünfte if the user has, say, Arbeitslosengeld or Bürgergeld?

**Review by Spencer6497 — CHANGES_REQUESTED** [review]
The code consolidation/reuse here is sensible, but I'm not sure the doneFunction logic is quite correct (unless I missed something?) Happy to discuss as always :)

File: app/flows/prozesskostenhilfeFormular/finanzielleAngaben/doneFunctions.ts (hunk 2 above)
Comment (rbrtrfl) [reply app/flows/prozesskostenhilfeFormular/finanzielleAngaben/doneFunctions.ts:19]: you are right! thats one staatliche Leistung too much :)

File: app/flows/prozesskostenhilfeFormular/abgabe/flow.json [hunk 6]
```diff
@@ -22,14 +34,22 @@
...
+          },
+          {
+            "guard": "staatlicheLeistungenIsBuergergeld",
+            "target": "#finanzielle-angaben.eigentum.kraftfahrzeuge-frage"
+          },
+          {
+            "guard": "hasNoStaatlicheLeistungen",
+            "target": "#ausgaben.ausgaben-frage"
           },
-          "#ausgaben"
```
[the remark this answers is not in the document]
Comment (rbrtrfl) [reply app/flows/prozesskostenhilfeFormular/abgabe/flow.json]: after my code changes, `hasStaatlicheLeistungen` and `hasNoStaatlicheLeistungen` is now again exclusively used by BerH. So I've changed the guards around the `BACK` route in PKH `#abgabe` 🤍

**Comment by rbrtrfl** [comment]
@Spencer6497 thanks for helping finish my WIP for this subtask 😉

File: app/flows/prozesskostenhilfeFormular/finanzielleAngaben/doneFunctions.ts [hunk 7]
```diff
@@ -66,23 +69,25 @@ export const wertsachenDone: ProzesskostenhilfeFinanzielleAngabenGuard = ({
...
-  eigentumDone({ context }) &&
-  kinderDone({ context }) &&
-  eigentumZusammenfassungDone({ context }) &&
-  andereUnterhaltszahlungenDone({ context }) &&
-  ausgabenDone({ context }) &&
-  ausgabenZusammenfassungDone({ context });
+export const prozesskostenhilfeFinanzielleAngabeDone: ProzesskostenhilfeFinanzielleAngabenGuard =
+  ({ context }) => {
+    return context.staatlicheLeistungen === "asylbewerberleistungen" ||
+      context.staatlicheLeistungen === "grundsicherung"
```
Comment (Spencer6497) [inline app/flows/prozesskostenhilfeFormular/finanzielleAngaben/doneFunctions.ts:75]: could use the more succinct `hasGrundsicherungOrAsylbewerberleistungen({
        context,
      }))` syntax but your choice :)

**Review by Spencer6497 — APPROVED** [review]
excellent work 🎉 🚢

**Pull request merged**